lib/common/compiler.h

```diff
@@ -63,6 +63,21 @@
 #  endif
 #endif
 
+/* target attribute */
+#if defined(__GNUC__)
+#  define TARGET_ATTRIBUTE(target) __attribute__((__target__(target)))
+#else
+#  define TARGET_ATTRIBUTE(target)
+#endif
+
+#ifndef DYNAMIC_BMI2
+  #if defined(__GNUC__) && (defined(__x86_64__) || defined(_M_X86))
```

Comment:
probably add `&& !defined(__BMI2__)`
as in this case, the base code path is already bmi2, so there is no need for a dispatch